api/api.js

```diff
@@ -1534,6 +1562,68 @@ export default (router) => {
     }
   })
 
+  router.get('/hubs/:publicKeyOrHandle/releases/archived', async (ctx) => {
+    try {
+      await NinaProcessor.init()
+      let { offset=0, limit=BIG_LIMIT, sort='desc', column='datetime' } = ctx.query;
+      column = formatColumnForJsonFields(column);
+      const hub = await hubForPublicKeyOrHandle(ctx)
+      let releases
+      const archivedReleasesForHub = await Release
+        .query()
+        .joinRelated('hubs')
+        .where('hubs_join.hubId', hub.id)
+        .where('hubs_join.visible', false)
+        .where(ref('metadata:name').castText(), 'ilike', `%${query}%`)
+        .orderBy(column, sort)
+        .range(Number(offset), Number(offset) + Number(limit) - 1);
+      releases = await Release.query()
+        .whereIn('id', archivedReleasesForHub.results.map(release => release.id))
+
+      releases = {
+        results: releases,
+        total: archivedReleasesForHub.total
+      }
+
+      const hubContentPublicKeys = []
+      for await (let release of releases.results) {
+        const [hubContentPublicKey] = await anchor.web3.PublicKey.findProgramAddressSync(
+          [
+            Buffer.from(anchor.utils.bytes.utf8.encode("nina-hub-content")), 
+            new anchor.web3.PublicKey(hub.publicKey).toBuffer(),
+            new anchor.web3.PublicKey(release.publicKey).toBuffer(),
+          ],
+          NinaProcessor.program.programId
+        )
+        hubContentPublicKeys.push(hubContentPublicKey)
+      }
+      const hubContent = await NinaProcessor.program.account.hubContent.fetchMultiple(hubContentPublicKeys, 'confirmed')
+      for await (let release of releases.results) {
+        const releaseHubContent = hubContent.filter(hc => hc.child.toBase58() === release.hubReleasePublicKey)[0]
+        if (releaseHubContent) {
+          release.datetime = new Date(releaseHubContent.datetime.toNumber() * 1000).toISOString()
+        }
+      }
+      
+      if (sort === 'desc') {
+        releases.results.sort((a, b) => new Date(b.datetime) - new Date(a.datetime))
+      } else {
+        releases.results.sort((a, b) => new Date(a.datetime) - new Date(b.datetime))
+      }
+
+      ctx.body = { 
+        releases: releases.results,
+        total: releases.total,
+        publicKey: hub.publicKey,
+        query,
```

Comment:
Is this actually supposed to be desconstructed from ctx.query()? Unless I'm missing something I don't see this defined anywhere.

Reply:
good - catch - removed it and forgot to add back